lgc/patch/PatchBufferOp.cpp

```diff
@@ -120,7 +120,9 @@ bool PatchBufferOp::runImpl(Function &function, PipelineState *pipelineState,
   m_pipelineState = pipelineState;
   m_isDivergent = std::move(isDivergent);
   m_context = &function.getContext();
-  m_builder = std::make_unique<IRBuilder<>>(*m_context);
+
+  IRBuilder<> builder(*m_context);
+  m_builder = &builder;
```

Comment:
On second thought, maybe using an `Optional<IRBuilder>` as class member is even cleaner, setting/resetting in the run function?

Reply:
Hmm. Having pass class members that are valid only for the duration of the `run` method is sort of an established pattern. Definitely using `Optional` here feels like it would cause more confusion than help. And yeah, we could do the make_scope_exit thing, but I'm sort of hesitant to establish a precedent here.

What we really should be doing more IMO is to lean on ephemeral classes/objects that live only for the duration of the run method, and then do all the real implementation in those. This pattern is used in a bunch of places in LLVM, it's just not used consistently.

Reply:
I agree, the latter would be cleaner. I've run into cases where it's not entirely clear that a second `run()` does the same as as a first one due to leaking state through members, which would also be eliminated by moving state into a temporary object doing the actual work.